### PR TITLE
Add additional configurable Options

### PR DESCRIPTION
**Pull request by spthiel**
This change aims to allow users to further customize their experience it aims for the following two changes:
1. ability to change the text of broadcast chat messages
2. ability to individually disable or enable controls in the hover menu

### CODE REVIEW

File: src/main/kotlin/tech/thatgravyboat/craftify/ui/UIControls.kt [hunk 1]
```diff
@@ -30,65 +31,59 @@ class UIControls : UIContainer() {
     private val position = "https://i.imgur.com/XZWUSSe.png"
     private val settings = "https://i.imgur.com/Nd4gQzY.png"
 
-    private val positionButton = UIButton(URL(position), URL(position), click = {
+    private val positionButton by UIButton(URL(position), URL(position), click = {
```
Comment (spthiel) [inline src/main/kotlin/tech/thatgravyboat/craftify/ui/UIControls.kt:34]: These are changed to by as per [elementa styleguide](https://github.com/EssentialGG/Elementa/blob/master/docs/style-guide.md#language)

File: src/main/kotlin/tech/thatgravyboat/craftify/config/Config.kt [hunk 2]
```diff
@@ -189,6 +189,14 @@ object Config : Vigilant(File("./config/craftify.toml")) {
     )
     var announceNewSong = 0
 
+    @Property(
+            type = PropertyType.TEXT,
+            name = "Announcement Message",
+            category = "General",
+            description = "Format for the announcement message (chat message)"
```
Comment (ThatGravyBoat) [inline src/main/kotlin/tech/thatgravyboat/craftify/config/Config.kt:196]: Provide what the user can input into a message for replacements just incase if they forget, also change the replacements to be `${variable}` instead.

File: src/main/kotlin/tech/thatgravyboat/craftify/ui/UIControls.kt [hunk 3]
```diff
@@ -122,39 +115,47 @@ class UIControls : UIContainer() {
...
+
+        if (ThemeConfig.showPositionEditorButton && (!ThemeConfig.hideImage || activeButtons < 7)) {
+            positionButton.updateImage(notNullNotBlankOrElse(ThemeConfig.positionEditorIcon, position))
+            positionButton.unhide()
+        } else  {
+            positionButton.hide(true)
+        }
+    }
+
+    private fun updateButton(button: UIButton, visible: Boolean, image: String, default: String, activeButtons: Int = 0): Int {
```
Comment (ThatGravyBoat) [inline src/main/kotlin/tech/thatgravyboat/craftify/ui/UIControls.kt:150]: change to allow secondary optional image and use for all buttons including volume, position, and play to keep consistency.

File: src/main/kotlin/tech/thatgravyboat/craftify/ui/Player.kt [hunk 4]
```diff
@@ -54,8 +54,9 @@ object Player {
         AdManager.changeAd()
         if (Config.announceNewSong == 1) {
             UChat.chat(
-                "${ChatColor.GREEN}Craftify > ${ChatColor.GRAY}" +
-                        "Now Playing: ${ChatColor.AQUA}${state.song.title} by ${state.song.artists.joinToString(", ")}"
+                Config.announcementMessage
+                        .replace("%SONG%", state.song.title)
```
Comment (ThatGravyBoat) [inline src/main/kotlin/tech/thatgravyboat/craftify/ui/Player.kt:58]: Same as config comment change to `${variable}` and keep it lowercase

**Review by ThatGravyBoat — CHANGES_REQUESTED** [review]

File: src/main/kotlin/tech/thatgravyboat/craftify/config/Config.kt (hunk 2 above)
Comment (spthiel) [reply src/main/kotlin/tech/thatgravyboat/craftify/config/Config.kt]: Is this what you intended?

**Review by ThatGravyBoat — APPROVED** [review]

**Pull request merged**